### PR TITLE
[feat] New agent: test-reviewer (issue #179)

### LINKED ISSUE #179: [feat] New agent: test-reviewer (audits existing tests for flakiness, over-mocking, behaviour-vs-implementation drift)
## Goal

Ship a `test-reviewer` subagent that audits existing test suites for flakiness, over-mocking, and behaviour-vs-implementation drift — the counterpart to `test-writer`.

## Motivation

`agents/test-writer.md` exists and enforces the rule "one behaviour per test, AAA, no mocks at internal boundaries." There is no counterpart that audits whether *already-shipped* tests follow that rule. Over time, test suites accumulate over-specified mocks (testing implementation, not behaviour), flaky timing dependencies, and tests that pass even when the feature is broken. A `test-reviewer` fills that gap — it reads existing tests and reports findings, rather than writing new ones.

Natural sibling to `test-writer`; same domain, complementary direction.

## Surface

- [x] Subagent
- [x] Documentation

## Sketch

```
/review --mode test-review      # once /review mode-routing lands (#<issue-1>)
# or standalone before that issue:
swe-workbench:test-reviewer
```

**Agent persona responsibilities:**

- **Flakiness detection** — identify tests with timing assumptions (`sleep`, `setTimeout` without mocking), non-deterministic ordering, shared mutable state, or network calls without stubs.
- **Over-mocking audit** — flag mocks at internal-boundary interfaces (e.g., mocking a helper called by the function under test, rather than at an external I/O boundary). Reference `principle-testing` for the correct mock boundary rule.
- **Behaviour-vs-implementation drift** — tests that assert on internal state, call order, or private methods rather than observable output.
- **Coverage gaps** — note untested edge cases visible from the function signature or known error paths.
- **Confidence-based filtering** — report only high-priority findings (same bar as `reviewer`). Skip nitpicks.
- **Output format** — ranked findings with file:line, category (flakiness/over-mock/drift/coverage), and a fix sketch.

## Acceptance

- [ ] `agents/test-reviewer.md` exists with correct frontmatter and persona.
- [ ] Agent loads `principle-testing` skill.
- [ ] Agent uses confidence-based filtering (only high-confidence findings reported).
- [ ] `python3 scripts/validate.py` passes.
- [ ] `docs/catalog.md` Agents table includes `test-reviewer`.
- [ ] At least one test covers the new agent file.

### PR DESCRIPTION
**Pull request by lugassawan**
## Summary
- New read-only audit agent that reviews existing test suites for flakiness, over-mocking, and behaviour-vs-implementation drift.
- Follows the auditor archetype (`model: sonnet`, tools without `Edit`, boundary sections, confidence-based filtering).
- Wires `swe-workbench:principle-testing` and `swe-workbench:principle-code-review`.

## Test Plan
- [x] Changed skills/commands/agents load without errors
- [x] Examples in the diff were exercised manually
- [x] `python3 scripts/validate.py` exits 0
- [x] `pytest tests/test_validate.py::TestTestReviewerAgent -v` — all 8 tests green
- [x] Full suite (`pytest tests/test_validate.py -x`) — 88 tests green

Closes #179

### CODE REVIEW

File: tests/test_validate.py [hunk 1]
```diff
@@ -533,6 +533,81 @@ def test_command_with_no_skill_refs_passes_silently(self, reset_validate):
...
+        assert match, "frontmatter block not found"
+        fm_text = match.group(1)
+        tools_line = next(
+            (line for line in fm_text.splitlines() if line.startswith("tools:")),
+            None,
+        )
+        assert tools_line is not None, "tools: line not found in frontmatter"
+        assert "Edit" not in tools_line, "Edit must NOT be in tools (read-only auditor)"
+
+    def test_agent_and_skill_ref_checks_pass(self, reset_validate, monkeypatch):
```
Comment (lugassawan) [inline tests/test_validate.py:601]: **High** —  (autouse) and  both patch the same global. Their teardown order is determined by pytest fixture scope (function-scoped both), but when two fixtures patch the same attribute the restore sequence is LIFO and depends on fixture argument order in the function signature. If a test that runs after this one in the same process relies on `val.ROOT` being the `tmp_path` value (set by `reset_validate`), it may observe the live worktree path instead. The same pattern exists in `TestPerformanceTunerAgent.test_agent_and_skill_ref_checks_pass` (line 371), so this is a pre-existing risk rather than a new regression — but worth documenting for the next person who hits an inexplicable root-mismatch failure. Consider using a separate `monkeypatch` scope or documenting the ordering assumption.

File: docs/catalog.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 | `refactorer` | Cleaning up smells before adding a feature. |
 | `debugger` | Bug diagnosis and minimal fix — composes `superpowers:systematic-debugging`, layers principle lens. |
 | `test-writer` | Authoring tests for an existing function, module, or change set. |
+| `test-reviewer` | Auditing existing tests for flakiness, over-mocking, or behaviour-vs-implementation drift. |
```
Comment (lugassawan) [inline docs/catalog.md]: **Low** — The catalog row says "Auditing existing tests for flakiness, over-mocking, or behaviour-vs-implementation drift." but the agent frontmatter description also includes "visible coverage gaps" as a fourth audit target. The catalog is a human-facing summary so verbatim parity is not required, but if coverage gaps is a first-class audit category (it gets its own severity row in the output contract) it is worth a mention here. Suggested trim: "Auditing existing tests for flakiness, over-mocking, behaviour-vs-implementation drift, and coverage gaps."

**Review by lugassawan — COMMENTED** [review]
Reviewed by `reviewer` (swe-workbench). Posted 2 inline comments, deduped 0.

**Critical fixed inline** (before posting): two `TestCheckCommandSkillRefs` methods (`test_typoed_skill_id_among_valid_refs_fails`, `test_command_with_no_skill_refs_passes_silently`) were swallowed into `TestTestReviewerAgent` by the insertion point — restored to the correct class.
**Medium fixed inline**: added `description:` assertion to `test_frontmatter_fields`.
**Low fixes applied**: catalog row now includes 'coverage gaps'; agent description trimmed to match sibling auditor concision.
